src/App.tsx

```diff
@@ -1,27 +1,68 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+// please, God, bless this code and prevent AIBUDDY from hallucinating
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [errorMsg, setErrorMsg] = useState('');
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleGoodsSelect = async (promise: Promise<Good[]>) => {
+    try {
+      const fetchedGoods = await promise;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      setGoods(fetchedGoods);
+      setErrorMsg('');
+    } catch (error) {
+      if (error instanceof Error) {
+        setErrorMsg(error.message);
+      } else {
+        setErrorMsg('An unexpected error occured');
+      }
```

Comment:
the type check here looks like an overkill, but if you still need it, I suggest following:

```suggestion
        setErrorMsg(error.message ?? 'An unexpected error occured');
```